Documentation/DomainModel/Entities.md

```diff
@@ -0,0 +1,235 @@
+
+A subscriber can have some **attributeValues**.
+
+### SubscriberHistory
+Table name: `phplist_user_user_history`
+
+This contains records of changes to a **subscriber** account. In phpList 3,
+the trail is accessible on the subscriber history page, e.g.,
+/admin/?page=userhistory&id=1. It includes when and how they were subscribed
+and reasons for their blacklisting.
+
+### Subscription
+Table name: `phplist_listuser`
+
+A **subscription** means that a **subscriber** is subscribed to a **list**.
+
+### Subscriber-Campaign Connection
+Table name: `phplist_usermessage`
+
+This association is the core of everything. A subscriber can be member of
+multiple lists, and a campaign can be sent to multiple lists, but this
+association ensures that a subscriber always only receives one copy of a
+campaign, regardless of other associations.
+
+Should we use a named association for this? What should it be named?
+
+### SuppressionList
+Table name: `phplist_user_blacklist`
+
+When a *subscriber* should be considered removed from the system, they are
+added here. Any email address mentioned should never receive any further
+emails, unless they actively go through the subscription and confirmation
+process again, when they will be removed from this list.
+
+### SuppressionListData
+
+Table name: `phplist_user_blacklist_data`
+
+This is some more additional info on a SuppressionList.
+
+
+## Messaging Context
+
+
+### Attachment
+Table name: `phplist_attachment`
+
+An attachment represents a file attached to exactly one **campaign**.
+
+### Bounce
+Table name: `phplist_boune`
+
+### BounceRegEx
+Table name: `phplist_bounceregex`
+
+### Campaign
+Table name: `phplist_message`
+
+A **campaign** is a non-personalized message on a **list** (or potentially
+multiple lists). The campaign has been created by an **administrator** owner
+and sends out email messages to multiple **subscribers**. It is stored to
+which subscribers a campaign has been sent.
+
+### CampaignBounce
+Table name: `phplist_message_bounce`
+
+### CampaignData
+Table name: `phplist_messagedata`
+
+This contains additional miscellaneous attributes for **campaigns**, such as
+Google tracking IDs, special relationships to **lists**, and alias titles.
+
+### CampaignForward
+Table name: `phplist_message_forward`
+
+This tracks details of **campaigns** which were forwarded by a recipient
+**subscriber** to someone else via an email message.
+
+### List
+Table name: `phplist_list`
+
+A **list** is something to which **subscribers** can subscribe in order to
+receive email messages from **campaigns**. Each list has exactly one **administrator**
+owner.
+
+### Template
+Table name: `phplist_template`
+
+This stores **campaign** templates. The blob contains HTML which the campaign
+content is inserted into, using special tags (AKA placeholders).
+
+### TemplateImage
+Table name: `phplist_templateimage`
+
+This contains images used in **templates**. The blob contains the image.
```

Comment:
template images are embedded in the email messages that are sent out, as opposed to content images, that are loaded when viewing the email message. As a result template images will show even when the recipient is offline.